python/ray/air/session.py

```diff
@@ -0,0 +1,246 @@
+    assert isinstance(
+        session, TrainSession
+    ), "`get_world_size` can only be called for TrainSession!"
+    return session.world_size
+
+
+def get_world_rank() -> int:
+    """Get the world rank of this worker.
+
+    .. code-block:: python
+
+        import time
+        from ray.air import session
+
+        def train_func():
+            for iter in range(100):
+                time.sleep(1)
+                if session.get_world_rank() == 0:
+                    print("Worker 0")
+
+        trainer = Trainer(backend="torch")
+        trainer.start()
+        trainer.run(train_func)
+        trainer.shutdown()
+    """
+    session = _get_session()
+    assert isinstance(
+        session, TrainSession
+    ), "`get_world_rank` can only be called for TrainSession!"
+    return session.world_rank
+
+
+def local_rank() -> int:
+    """Get the local rank of this worker (rank of the worker on its node).
+
+    .. code-block:: python
+
+        import time
+        from ray.air import session
+
+        def train_func():
+            if torch.cuda.is_available():
+                torch.cuda.set_device(session.get_local_rank())
+            ...
+
+        trainer = Trainer(backend="torch", use_gpu=True)
+        trainer.start()
+        trainer.run(train_func)
+        trainer.shutdown()
+    """
+    session = _get_session()
+    assert isinstance(
+        session, TrainSession
+    ), "`get_local_rank` can only be called for TrainSession!"
+    return session.local_rank
+
+
+def get_dataset_shard(
+    dataset_name: Optional[str] = None,
+) -> Optional[Union["Dataset", "DatasetPipeline"]]:
+    """Returns the Ray Dataset or DatasetPipeline shard for this worker.
+
+    You should call ``to_torch()`` or ``to_tf()`` on this shard to convert
+    it to the appropriate framework-specific Dataset.
+
+    .. code-block:: python
+
+        import ray
+        from ray import train
+        from ray.air import session
+
+        def train_func():
+            model = Net()
+            for iter in range(100):
+                data_shard = session.get_dataset_shard().to_torch()
+                model.train(data_shard)
+            return model
+
+        dataset = ray.data.read_csv("train.csv")
+        dataset.filter(...).repeat().random_shuffle()
+
+        trainer = Trainer(backend="torch")
+        trainer.start()
+
+        # Trainer will automatically handle sharding.
+        train_model = trainer.run(train_func, dataset=dataset)
+        trainer.shutdown()
+
+    Args:
+        dataset_name: If a Dictionary of Datasets was passed to ``Trainer``, then
+            specifies which dataset shard to return.
+
+    Returns:
+        The ``Dataset`` or ``DatasetPipeline`` shard to use for this worker.
+        If no dataset is passed into Trainer, then return None.
+    """
+    session = _get_session()
+    assert isinstance(
+        session, TrainSession
+    ), "`get_dataset_shard` can only be called for TrainSession!"
```

Comment:
nit: Instead of assertions we should raise proper RuntimeErrors here (and maybe be a bit more actionable in the message)